core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartDownscaleGetRunningInstancesHandler.java

```diff
@@ -0,0 +1,127 @@
+    private StackService stackService;
+
+    @Inject
+    private RuntimeVersionService runtimeVersionService;
+
+    @Inject
+    private CloudInstanceIdToInstanceMetaDataConverter cloudInstanceIdToInstanceMetaDataConverter;
+
+    @Inject
+    private InstanceMetaDataToCloudInstanceConverter instanceMetaDataToCloudInstanceConverter;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public Class<StopStartDownscaleGetRunningInstancesRequest> type() {
+        return StopStartDownscaleGetRunningInstancesRequest.class;
+    }
+
+    @Override
+    public void accept(Event<StopStartDownscaleGetRunningInstancesRequest> event) {
+        StopStartDownscaleGetRunningInstancesRequest request = event.getData();
+        LOGGER.info("StopStartDownscaleGetRunningInstancesHandler: {}", event.getData().getResourceId());
+
+        CloudContext cloudContext = request.getCloudContext();
+
+        try {
+            CloudConnector<?> connector  = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = getAuthenticatedContext(request, cloudContext, connector);
+
+            Stack stack = stackService.getByIdWithLists(request.getResourceId());
+            List<InstanceMetaData> startedInstancesMetadata = cloudInstanceIdToInstanceMetaDataConverter.getNotDeletedAndNotZombieInstances(
+                    stack, request.getHostGroupName(), collectStartedInstancesFromCloudProvider(connector, ac, request.getAllInstancesInHostGroup()).stream()
+                            .map(CloudInstance::getInstanceId)
+                            .collect(Collectors.toSet()));
+
+            ClusterStatusService clusterStatusService = clusterApiConnectors.getConnector(stack).clusterStatusService();
+
+            ExtendedHostStatuses extendedHostStatuses = clusterStatusService.getExtendedHostStatuses(
+                    runtimeVersionService.getRuntimeVersion(stack.getCluster().getId()));
+
+            List<InstanceMetaData> startedInstanceMetadataWithServicesNotRunning = startedInstancesMetadata.stream()
+                    .filter(i -> i.getInstanceGroupName().equals(request.getHostGroupName()))
+                    .filter(i -> !extendedHostStatuses.isHostHealthy(hostName(i.getDiscoveryFQDN())))
+                    .collect(Collectors.toList());
+
+            List<CloudInstance> instancesWithServicesNotRunning = instanceMetaDataToCloudInstanceConverter.convert(
+                    startedInstanceMetadataWithServicesNotRunning, stack.getEnvironmentCrn(), stack.getStackAuthentication());
+            StopStartDownscaleGetRunningInstancesResult result = new StopStartDownscaleGetRunningInstancesResult(stack.getId(),
+                    instancesWithServicesNotRunning, request.getHostGroupName(), request.getHostIds());
+            notify(result, event);
+        } catch (Exception e) {
+            String errorMessage = "Failed to collect started instances with unhealthy services from cloud provider";
```

Comment:
This message is misleading, since the error could occur from CM as well. i.e. Source system is unknown.